### PR TITLE
feat: enable parallel storage root computation on canonical chain write

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

File: crates/blockchain-tree/src/blockchain_tree.rs [hunk 1]
```diff
@@ -1151,16 +1153,13 @@ where
...
-                    .disable_long_read_transaction_safety();
-                let (state_root, trie_updates) = hashed_state
-                    .state_root_with_updates(provider.tx_ref())
-                    .map_err(Into::<BlockValidationError>::into)?;
+                let consistent_view =
+                    ConsistentDbView::new_with_latest_tip(self.externals.provider_factory.clone())?;
+                let (state_root, trie_updates) =
+                    ParallelStateRoot::new(consistent_view, hashed_state.clone())
+                        .incremental_root_with_updates()
+                        .map(|(root, updates)| (root, Some(updates)))
```
Comment (rkrasiuk) [inline crates/blockchain-tree/src/blockchain_tree.rs:1161]: no need to map here

File: crates/blockchain-tree/src/blockchain_tree.rs [hunk 2]
```diff
@@ -1151,16 +1153,13 @@ where
...
+                        .map_err(ProviderError::from)?;
```
Comment (rkrasiuk) [inline crates/blockchain-tree/src/blockchain_tree.rs:1162]: this should be BlockValidationError as before

File: crates/blockchain-tree/src/blockchain_tree.rs [hunk 3]
```diff
@@ -1151,16 +1153,13 @@ where
             }
             None => {
                 debug!(target: "blockchain_tree", blocks = ?block_hash_numbers, "Recomputing state root for insert");
-                let provider = self
-                    .externals
-                    .provider_factory
-                    .provider()?
-                    // State root calculation can take a while, and we're sure no write transaction
-                    // will be open in parallel. See https://github.com/paradigmxyz/reth/issues/6168.
-                    .disable_long_read_transaction_safety();
```
Comment (rkrasiuk) [inline crates/blockchain-tree/src/blockchain_tree.rs:1160]: worth configuring this in the consistent view

**Review by rkrasiuk — COMMENTED** [review]
looking good

**Comment by gakonst** [comment]
gm @fgimenez next steps here?

**Comment by Rjected** [comment]
bump @rkrasiuk for review

**Pull request closed without merging**